internal/pkg/vm/actor/actor.go

```diff
@@ -67,14 +70,34 @@ func (a *Actor) IncrementSeqNum() {
 // Cid returns the canonical CID for the actor.
 // TODO: can we avoid returning an error?
 func (a *Actor) Cid() (cid.Cid, error) {
-	obj, err := cbor.WrapObject(a, types.DefaultHashFunction, -1)
+	bs, err := encoding.Encode(a)
 	if err != nil {
 		return cid.Undef, errors.Wrap(err, "failed to marshal to cbor")
 	}
+	obj, err := cbor.Decode(bs, types.DefaultHashFunction, -1)
+	if err != nil {
+		return cid.Undef, errors.Wrap(err, "failed to cast cbor bytes to ipld node")
+	}
 
 	return obj.Cid(), nil
 }
 
+// GetHead returns the cid of the actor's storage, or Undef if unset
+func (a *Actor) GetHead() cid.Cid {
```

Comment:
Why are we adding Getters with `Get*` we are not using the `Get*` pattern on other data structures.

Reply:
Its an artifact of cid.Undef not being encodable / decodable forcing us to use datastructures with pointers to cid objects and therefore causing a need for non-nil dereferencing accessors.

Reply:
addressed